framework/test/gtest/GlobalServiceTrackerTest.cpp

```diff
@@ -0,0 +1,60 @@
+/*=============================================================================
+
+Library: CppMicroServices
+
+Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+file at the top-level directory of this distribution and at
+https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=============================================================================*/
+
+#include "cppmicroservices/GlobalConfig.h"
+
+#include "cppmicroservices/Framework.h"
+#include "cppmicroservices/FrameworkEvent.h"
+#include "cppmicroservices/FrameworkFactory.h"
+#include <cppmicroservices/ServiceInterface.h>
+#include "cppmicroservices/ServiceTracker.h"
+
+#include "gtest/gtest.h"
+
+#include <memory>
+
+// A dummy interface to use with service trackers
+namespace {
+    class Foo {
+    public:
+        virtual ~Foo() {}
```

Comment:
= default;

In fact, the destructor does not have to be virtual. This class is for test purpose only and there is no subclass for it.